base/pkg/entry.jl

```diff
@@ -125,6 +125,7 @@ function installed(pkg::AbstractString)
 end
 
 function status(io::IO; pkgname::AbstractString = "")
+    if ispath(pkgname) || throw(PkgError("package $pkgname does not exist")) end
```

Comment:
The best way to conditionally throw exceptions like this is a little simpler; just omit `if` and `end`. For example,
```julia
ispath(pkgname) || throw(PkgError("Package $pkgname does not exist"))
```